nmstatectl/nmstatectl.py

```diff
@@ -22,24 +22,40 @@
 import argparse
 import json
 
+from libnmstate import netapplier
 from libnmstate import netinfo
 
 
 def main():
     parser = argparse.ArgumentParser()
 
-    setup_subcommand_show(parser)
+    subparsers = parser.add_subparsers()
+    setup_subcommand_show(subparsers)
+    setup_subcommand_set(subparsers)
 
     args = parser.parse_args()
-    args.func()
+    args.func(args)
 
 
-def setup_subcommand_show(parser):
-    subparsers = parser.add_subparsers(help='Show network state')
-    parser_show = subparsers.add_parser('show')
+def setup_subcommand_show(subparsers):
+    parser_show = subparsers.add_parser('show', help='Show network state')
     parser_show.set_defaults(func=show)
 
 
-def show():
+def setup_subcommand_set(subparsers):
+    parser_set = subparsers.add_parser('set', help='Set network state')
+    parser_set.add_argument('-f', '--file',
+                            help='File containing desired state')
+    parser_set.set_defaults(func=apply)
```

Comment:
why don't you use the same name "set", but "apply" instead?

Reply:
Because "set" is reserved for dicts.